src/Dibi/Drivers/MySqliDriver.php

```diff
@@ -148,20 +148,26 @@ public function query(string $sql): ?Dibi\ResultDriver
 	}
 
 
+	/**
+	 * @param int|string $code
+	 */
 	public static function createException(string $message, $code, string $sql): Dibi\DriverException
 	{
-		if (in_array($code, [1216, 1217, 1451, 1452, 1701], true)) {
-			return new Dibi\ForeignKeyConstraintViolationException($message, $code, $sql);
+		if (is_int($code) === true) {
+			if (in_array($code, [1216, 1217, 1451, 1452, 1701], true)) {
```

Comment:
there is no reason for this change, in_array checks type

Reply:
Parameter `$code` can be `int` or `string`. In the case of `string` the `if` statements can be skipped.

![Snímek obrazovky 2020-05-07 v 22 35 17](https://user-images.githubusercontent.com/4738758/81342240-3753b580-90b3-11ea-8a25-217c28204215.png)


What do you think?

Reply:
I think in mysql driver it is always int.